examples/tutorials/tutorial_functions.py

```diff
@@ -0,0 +1,261 @@
+            bgr_embed_image = embed_image[..., 0:3][..., ::-1]
+
+        resized_bgr_embed_image = cv2.resize(
+            bgr_embed_image, thumb_size, interpolation=cv2.INTER_AREA
+        )
+
+        result_image[
+            y_offset - 1 : y_offset + outline_frame.shape[0] - 1,
+            x_offset - 1 : x_offset + outline_frame.shape[1] - 1,
+        ] = outline_frame
+        result_image[
+            y_offset : y_offset + resized_bgr_embed_image.shape[0],
+            x_offset : x_offset + resized_bgr_embed_image.shape[1],
+        ] = resized_bgr_embed_image
+        # move along y axis
+        y_offset = y_offset + y_bound + thumb_size[1]
+
+    return result_image
+
+
+def clip_depth_image(depth_image, clip_max=10.0):
+    """
+    Manually normalize depth into [0, 1] so that images are always consistent
+
+    Parameters
+    ----------
+    depth_image : 2d numpy array
+        Depth image data
+
+    Returns
+    -------
+    bgr_d_im : 2d numpy array
+        Clipped depth image data
+
+    """
+    d_im = np.clip(depth_image, 0, clip_max)
+    d_im /= clip_max
+    bgr_d_im = cv2.cvtColor((d_im * 255).astype(np.uint8), cv2.COLOR_GRAY2BGR)
+    return bgr_d_im
+
+
+def get_obs(sim, camera, show=True, save=False, output_path=""):
+    """
+    Get an agent observation for a specified camera.
+
+    Parameters
+    ----------
+    sim : Simulator, 
+        Reference to the habitat simulator object
+        
+    camera : string
+        Camera name to get observation from.
+        
+    show : boolean, optional
+        Whether to display an image of the observation. 
+        The default is True
+        
+    save : boolean optional
+        Whether to save image to path specified.  
+        The default is False
+        
+    output_path : string, optional
+        Where to save the image. The default is "".
+
+    Returns
+    -------
+    obs : list
+        2d numpy array of ints or floats representing observation.
+
+    """
+    obs = sim.get_sensor_observations()[camera]
+    if show:
+        show_img(obs, save, output_path)
+    return obs
+
+
+def show_img(data, save=False, output_path=""):
+    """
+    Display an image of the passed data
+
+    Parameters
+    ----------
+    data : list
+        Array-like structure of image data.  
+        See matplotlib.pyplot.imshow for more.
+    save : boolean optional
+        Whether to save image to path specified
+    output_path : string, optional
+        DESCRIPTION. The default is "".
+
+    Returns
+    -------
+    None.
+
+    """
+    from matplotlib import pyplot as plt
+
+    plt.figure(figsize=(12, 12))
+    plt.imshow(data, interpolation="nearest")
+    plt.axis("off")
+    plt.show(block=False)
+    if save:
+        global save_index
+        plt.savefig(
+            output_path + str(save_index) + ".jpg",
+            bbox_inches="tight",
+            pad_inches=0,
+            quality=50,
+        )
+        save_index += 1
+    plt.pause(1)
```

Comment:
Why is the pause for 1 second necessary here.

Reply:
Not sure, I just refactored this functionality from an existing tutorial, and I do not know off the top of my head whether this is required or not.